### PR TITLE
Add exemplar support

### PR DESCRIPTION
**Pull request by janstenpickle**
Exemplars allow us to attach transient labels like trace IDs to metrics

Note that some of this is still WIP, especially `Timer` and `OutcomeRecorder`. We'll be doing some more refactoring before coming back for those

### CODE REVIEW

**Review by alejandrohdezma — APPROVED** [review]
Looks great to me! 👏🏼 👏🏼 

Left some minor comments

One of the things I was thinking is, since this is going into a 2.0 milestone, we should embed exemplar support into every metric type that supports it, and easily provide adding an implicit noop that anyone can import. It would greatly simplify the codebase.

File: core/src/main/scala/prometheus4cats/Exemplar.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  /** Refined value class for an exemplar label name that has been parsed from a string
+    */
+  final class LabelName private (val value: String) extends AnyVal with internal.Refined.Value[String] {
+    override def toString: String = s"""Exemplar.LabelName("$value")"""
+  }
+
+  object LabelName extends internal.Refined.StringRegexRefinement[LabelName] with ExemplarLabelNameFromStringLiteral {
+    protected val regex: Pattern = "^[a-zA-Z_][a-zA-Z_0-9]*$".r.pattern
+
+    implicit val ordering: Ordering[LabelName] = catsInstances.toOrdering
```
[the remark this answers is not in the document]
Comment (janstenpickle) [reply core/src/main/scala/prometheus4cats/Exemplar.scala:51]: Yep it does, but we're going the other way as `SortedMap` needs `Ordering`

File: core/src/main/scala/prometheus4cats/Exemplar.scala [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    override protected def make(a: String): LabelName = new LabelName(a)
```
[the remark this answers is not in the document]
Comment (janstenpickle) [reply core/src/main/scala/prometheus4cats/Exemplar.scala:53]: That comes from our internal refinement code

File: core/src/main/scala/prometheus4cats/MetricRegistry.scala [hunk 3]
```diff
@@ -478,14 +480,218 @@ trait MetricRegistry[F[_]] {
       help: Metric.Help
   ): Resource[F, Info[F, Map[Label.Name, String]]]
 
-  final def mapK[G[_]](fk: F ~> G)(implicit F: MonadCancel[F, _], G: MonadCancel[G, _]): MetricRegistry[G] =
+  /** Create and register an exemplar counter that records [[scala.Double]] values against a metrics registry
```
[the remark this answers is not in the document]
Comment (janstenpickle) [reply core/src/main/scala/prometheus4cats/MetricRegistry.scala]: Resolving as this method no longer exists :D

**Review by alejandrohdezma — APPROVED** [review]

**Pull request merged**